wordpress.org/public_html/wp-content/plugins/plugin-directory/readme/class-parser.php

```diff
@@ -408,13 +416,18 @@ protected function parse_readme_contents( $contents ) {
 		// Use the first line of the description for the short description if not provided.
 		if ( ! $this->short_description && ! empty( $this->sections['description'] ) ) {
 			$this->short_description = array_filter( explode( "\n", $this->sections['description'] ) )[0];
+			$this->warnings['no_short_description_present'] = true;
 		}
 
 		// Sanitize and trim the short_description to match requirements.
 		$this->short_description = $this->sanitize_text( $this->short_description );
 		$this->short_description = $this->parse_markdown( $this->short_description );
 		$this->short_description = wp_strip_all_tags( $this->short_description );
-		$this->short_description = $this->trim_length( $this->short_description, 150 );
+		$short_description = $this->trim_length( $this->short_description, 150 );
```

Comment:
At this point `$this->short_description` already fallback to `$this->sections['description']`. So we are still not sure what is the actual length of the original short description.

Reply:
Resolving that is not worth the effort.

Instead, it should be checked something like this instead then:
```php
isset( $this->warnings['trimmed_short_description'] ) && ! isset( $this->warnings['no_short_description_present'] )
```

Reply:
Alternatively, we could avoid setting it, based on if a short description was provided, see below suggestion.